src/Bonus.java

```diff
@@ -1,66 +1,59 @@
 import java.util.HashSet;
 import java.util.Objects;
+import java.util.Random;
 import java.util.Scanner;
 
 public class Bonus {
 
     public static void main(String[] args) {
 
-        // Onderstaande code is uitgecommentarieerd. Haal de code uit de comment (delete /* en */) om aan de slag te gaan met de bonus
-        // In de code staan op verschillende plekken comments tussen "/*...*/" als hint wat je moet doen
-
-
-        HashSet<Integer> secretnumber = randomnumbergenerator();
-        String stringnumber = setToStringConverter(secretnumber);
-        System.out.println(stringnumber);
-        feedback(/*vul hier het juiste argument in*/);
-
+        HashSet<Integer> secretNumber = randomnumbergenerator();
+        String stringNumber = setToStringConverter(secretNumber);
+        // System.out.println(stringNumber);
+        feedback(stringNumber);
     }
 
-    public static void/*moet dit returntype "void" zijn of wat anders?*/ randomnumbergenerator(/*Heeft deze methode nog parameter(s) nodig?*/){
-         /*
-        Vul hier de body van de methode in.
-
-        Stappenplan:
-        - Maak een nieuwe variabele van type Random. (Tip: Zoek op internet hoe je Random kunt gebruiken)
-        - Maak een nieuwe variabele van type HashSet.
-        - Schrijf een while-loop om 4 random nummers aan de hashset toe te voegen
-        - return de hashset
-         */
+    public static HashSet<Integer> randomnumbergenerator() {
+        Random random = new Random();
+        HashSet<Integer> set = new HashSet<>();
+        while (set.size() < 4) {
+            set.add(random.nextInt(10));
```

Comment:
Sorry, ben aan het _nitpicken_ denk ik, maar as ik het goed begrijp worden er hier 9 getallen gegenereerd en speel je een standaard spel Mastermind met acht kleuren. (Ik dacht eerst trouwens dat er 10 werden gemaakt met deze Random klasse. Copilot kwam met het volgende over mijn observatie:

5. **Mastermind getallenbereik**:
   - **Observatie**: De methode `random.nextInt(10)` genereert getallen van 0 tot 9, wat betekent dat 10 niet wordt meegenomen.
   - **Aanbeveling**: Als de officiële Mastermind slechts 8 getallen gebruikt, pas de methode aan naar `random.nextInt(8)`.

Reply:
🤖 Als je getallen **tussen 1 en 8** (inclusief) wilt genereren in plaats van **0-7**, moet je de `nextInt(bound)`-methode van `Random` aanpassen. De methode `random.nextInt(8)` genereert getallen van **0 t/m 7**, dus je moet **1** optellen bij de gegenereerde waarde.

Hier is je aangepaste code:

### Aangepaste Code:
```java
public static List<Integer> randomNumberGenerator() {
    Random random = new Random();
    List<Integer> list = new ArrayList<>();
    while (list.size() < 4) {
        int nextNumber = random.nextInt(8) + 1; // Genereer getal tussen 1 en 8
        if (!list.contains(nextNumber)) {
            list.add(nextNumber);
        }
    }
    return list;
}
```

### Uitleg:
- **`random.nextInt(8)`**: Genereert getallen van **0 t/m 7**.
- **`+ 1`**: Verschuift het bereik naar **1 t/m 8**.

### Waarom werkt dit?
Het bereik wordt door `nextInt(bound)` bepaald als `0` (inclusief) tot `bound` (exclusief). Door er `1` bij op te tellen, wordt het resultaat:
- 0 → 1  
- 1 → 2  
- ...  
- 7 → 8  

---

### Resultaat:
De lijst bevat nu 4 unieke getallen tussen **1 en 8** (inclusief).